crates/autopilot/src/maintenance.rs

```diff
@@ -86,17 +87,47 @@ impl Maintenance {
     }
 
     async fn update_inner(&self) -> Result<()> {
-        // All these can run independently of each other.
-        tokio::try_join!(
-            self.settlement_indexer.run_maintenance(),
-            self.db_cleanup.run_maintenance(),
-            self.index_ethflow_orders(),
+        let settlement_indexer_fut = async {
```

Comment:
This is a lot of duplication that could be avoided by a simple helper function:
```rust
/// Runs the future and collects runtime metrics.
async fn timed_future<T>(label: &str, fut: impl Future<Output = T>) -> T {
    let _timer = metrics()
        .maintenance_stage_time
        .with_label_values(&[label])
        .start_timer();
    fut.await
}
```

Then you can do:
```rust
        tokio::try_join!(
            timed_future(
                "settlement_indexer",
                self.settlement_indexer.run_maintenance()
            ),
            timed_future("db_cleanup", self.db_cleanup.run_maintenance()),
            timed_future("ethflow_indexer", self.index_ethflow_orders()),
            timed_future("cow_amm_indexer", index_cow_amms),
        )?;
```